community/kernel/src/main/java/org/neo4j/kernel/impl/transaction/log/CommandWriter.java

```diff
@@ -293,11 +293,19 @@ public boolean visitRelationshipCountsCommand( RelationshipCountsCommand command
 
     private void writeMap( Map<String,Integer> map ) throws IOException
     {
-        channel.put( (byte) map.size() );
+        if ( map.size() > IndexDefineCommand.HIGHEST_POSSIBLE_ID )
+        {
+            throw new AssertionError( "Can not write map with size larger than 2 bytes. Actual size " + map.size() );
```

Comment:
mm...the only thing that i think is important is to use some kind of exception instead of Error, since it's just unsupported case and not an error that should cause db halt for example